lib/cib/cib_remote.c

```diff
@@ -210,9 +211,34 @@ cib_remote_callback_dispatch(gpointer user_data)
 
     xmlNode *msg = NULL;
 
-    crm_info("Message on callback channel");
+    /* If start_time is 0, we've previously handled a complete message and this
+     * callback is being reused for a new message.  Reset the start_time, giving
+     * this new message timeout_sec from now to complete.
+     */
+    if (rd->start_time == 0) {
+        rd->start_time = time(NULL);
+    } else if (time(NULL) >= rd->start_time + rd->timeout_sec) {
+        crm_info("Error reading from remote client: %s", pcmk_rc_str(ETIME));
```

Comment:
s/remote client/CIB manager connection/

This is the client reading from the remote (from its point of view) server

I wonder whether we should check the timeout only in the EAGAIN case -- if we can read a whole message now, might as well consider it good